test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -131,7 +131,9 @@ def do_h(int[32] q) {
 }
 def recursive_h(int[32] q) {
     do_h(q);
-    if (q > 0) {
+    bool __bool_0__;
```

Comment:
@rmshaffer one of your PRs simplified code like this recently -- should I investigate this more and see if I can get simplified output from this?

Reply:
I believe this is happening because the assignments converter is running after your new comparisons converter. I wonder if just moving your new `comparisons.transform()` call to happen _after_ the `assignments.transform()` call in `transpiler.py` would fix this.

Reply:
Hm I moved it down below logical expressions, but no effect on the test output

Reply:
Ok, interesting. I wonder if to fix this somehow your comparison converters would have to return oqpy expressions rather than variables - if you're declaring a variable I don't see how to avoid this. Anyway, it doesn't seem important for this PR (and maybe not important at all).

Reply:
Might be nice to revisit in Q1 -- I will make an issue